Sources/Testing/Testing.docc/Expectations.md

```diff
@@ -10,15 +10,54 @@ See https://swift.org/LICENSE.txt for license information
 See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 -->
 
-Check for expected values and outcomes in tests.
+Check for expected values, outcomes, and events in tests.
 
 ## Overview
 
-The testing library provides `#expect()` and `#require()` macros you use to 
-validate expected outcomes. To validate that an error is thrown, or _not_ thrown, 
-the testing library provides several overloads of the macros that you can use.
-Use a ``Confirmation`` to confirm the occurrence of an asynchronous event that
-you can't check directly using an expectation.
+Use ``expect(_:_:sourceLocation:)`` and
+``require(_:_:sourceLocation:)-5l63q`` macros to validate expected
+outcomes.  To validate that an error is thrown, or _not_ thrown, the
+testing library provides several overloads of the macros that you can
+use.  Use a ``Confirmation`` to confirm the occurrence of an
+asynchronous event that you can't check directly using an expectation.
+
+### Validate your code's result
+
+To validate that your code produces an expected value, use
+``expect(_:_:sourceLocation:)``.  ``expect(_:_:sourceLocation:)`` captures the
+expression you pass, and provides detailed information when the code doesn't
+satisfy the expectation:
+
+```swift
+class OrderCalculator {
+    func total(of subtotals: [Int]) -> Int {
+        return subtotals.reduce(1) { partialResult, subtotal in
+            partialResult + subtotal
+        }
+    }
+}
+
+@Test func calculatingOrderTotal() {
+    let calculator = OrderCalculator()
+    #expect(calculator.total(of: [3, 3]) == 6)
+    // Prints "Expectation failed: (calculator.total(of: [3, 3]) → 7) == 6"
+}
+```
+
+Your test keeps running after ``expect(_:_:sourceLocation:)`` fails.  To stop
+the test when the code doesn't satisfy a requirement, use
+``require(_:_:sourceLocation:)-5l63q`` instead:
+
+```swift
+@Test func returningCustomerRemembersUsualOrder() throws {
+    let customer = Customer(id: 123)
+    try #require(customer)
+    #expect(customer?.usualOrder?.countOfItems == 2) // The test runner doesn't reach this line if the customer is nil.
```

Comment:
This comment may trigger a horizontal scrollbar. How about:

```suggestion
    // The test runner doesn't reach this line if the customer is nil.
    #expect(customer?.usualOrder?.countOfItems == 2)
```

Reply:
Note if you apply my suggestion for the above line, `customer?` here becomes `customer`.